cmd/primaza/main.go

```diff
@@ -162,17 +166,44 @@ func main() {
 		os.Exit(1)
 	}
 
+	var hci int
+	if hci, err = strconv.Atoi(cfg.HealthCheckInterval); err != nil {
+		hci = 600
+		setupLog.Error(err, "default health check interval not set")
+	}
+
+	if err := mgr.Add(&clusterEnvironmentHealthcheck{
+		cer: cer,
+		ns:  cfg.WatchNamespace,
+		hci: hci,
+	}); err != nil {
+		setupLog.Error(err, "unable to set up ClusterEnvironments healthchecks")
+		os.Exit(1)
+	}
+
 	setupLog.Info("starting manager")
 	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
 		setupLog.Error(err, "problem running manager")
 		os.Exit(1)
 	}
 }
 
+type clusterEnvironmentHealthcheck struct {
+	cer *controllers.ClusterEnvironmentReconciler
+	ns  string
+	hci int
+}
+
+func (h *clusterEnvironmentHealthcheck) Start(ctx context.Context) error {
+	go h.cer.MonitorHealth(ctx, h.ns, h.hci)
+	return nil
+}
+
```

Comment:
I'd propose to move this in `pkg/primaza/healthchecks/clusterenvironment_healthchecker.go`